firedrake/mesh.py

```diff
@@ -2010,8 +2010,10 @@ def spatial_index(self):
         # tolerance. Note that if tolerance is too small it might not actually
         # change the value!
         if hasattr(self, "tolerance") and self.tolerance is not None:
```

Comment:
If we have no tolerance, then we don't draw our new kind of bounding box. Is that the desired behaviour?

Reply:
You are right. I've fixed this.